indico/modules/events/registration/settings.py

```diff
@@ -0,0 +1,63 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from indico.modules.events.registration.models.items import PersonalDataType
+from indico.modules.events.settings import EventSettingsProxy
+
+
+class RegistrationSettingsProxy(EventSettingsProxy):
+    """Store per-event registration settings"""
+    def get_participant_list_columns(self, event, form=None):
+        if form is None:
+            # Columns form merged forms
+            return self.get(event, 'participant_list_columns')
+        else:
+            try:
+                # The int values are automatically converted to str when saved as JSON
+                return [int(column_id) for column_id in self.get(event, 'participant_list_form_columns')[str(form.id)]]
```

Comment:
I think this is much more readable when split into two lines:

    form_columns = self.get(event, 'participant_list_form_columns')[unicode(form.id)]
    return map(int, form_columns)